torchrl/collectors/collectors.py

```diff
@@ -322,10 +322,56 @@ def __init__(
         self._tensordict.set(
             "step_count", torch.zeros(*self.env.batch_size, 1, dtype=torch.int)
         )
-        self._tensordict_out = TensorDict(
-            {},
-            batch_size=[*self.env.batch_size, self.frames_per_batch],
-            device=self.passing_device,
+
+        # TODO: can we rely on policy having spec and ditch hasattr?
```

Comment:
I think we can assume that the policy has spec, but those spec can be:
None -> do a rollout
CompositeSpec() -> do a rollout
CompositeSpec(action=None) -> do a rollout
CompositeSpec(action=foo, hidden=None) -> do a rollout
CompositeSpec(action=foo) -> use spec
CompositeSpec(action=foo, hidden=bar) -> use spec